sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs

```diff
@@ -930,5 +936,72 @@ async Task ProcessMessage(ProcessMessageEventArgs args)
                 await processor.StopProcessingAsync();
             }
         }
+
+        [Test]
+        public async Task StopProcessingDoesNotResultInRedeliveredMessages()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false))
+            {
+                await using var client = CreateClient();
+                await using var sender = client.CreateSender(scope.QueueName);
+
+                int receivedCount = 0;
+                int messageCount = 10;
+
+                var sendTask = SendMessages();
+                await ReceiveMessages(true);
+                await ReceiveMessages(false);
+                await sendTask;
+                Assert.AreEqual(messageCount, receivedCount);
+
+                async Task SendMessages()
+                {
+                    for (int i = 0; i < messageCount; i++)
+                    {
+                        await sender.SendMessageAsync(ServiceBusTestUtilities.GetMessage());
+                        // send in 1 sec increments to intersperse receives and sends
+                        await Task.Delay(1000);
+                    }
+                }
+
+                async Task ReceiveMessages(bool first)
+                {
+                    bool received = false;
+                    await using var processor = client.CreateProcessor(scope.QueueName, new ServiceBusProcessorOptions
+                    {
+                        MaxConcurrentCalls = 5,
+                    });
+                    TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                    processor.ProcessErrorAsync += ServiceBusTestUtilities.ExceptionHandler;
+                    processor.ProcessMessageAsync += async args =>
+                    {
+                        await args.CompleteMessageAsync(args.Message);
+
+                        // add a 2 second delay to give the receive link more time to have messages delivered
+                        await Task.Delay(2000);
+
+                        try
+                        {
+                            Assert.AreEqual(1, args.Message.DeliveryCount);
+                        }
+                        catch (Exception ex)
+                        {
+                            tcs.TrySetException(ex);
+                        }
+
+                        int ct = Interlocked.Increment(ref receivedCount);
```

Comment:
```suggestion
                        int count = Interlocked.Increment(ref receivedCount);
```

![](https://media2.giphy.com/media/qpCvOBBmBkble/giphy.gif)